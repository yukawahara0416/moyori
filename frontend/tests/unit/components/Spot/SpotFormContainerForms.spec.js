import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Component from '@/components/Spot/SpotFormContainerForms.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper

let post
let store

beforeEach(() => {
  post = {
    namespaced: true,
    getters: {
      spotFormData: () => ({
        address: 'address',
        name: 'name',
        image: 'image',
        place_id: 'place_id',
        lat: '36.204824',
        lng: '138.252923',
        url: 'url'
      })
    },
    actions: {
      postSpot: jest.fn()
    }
  }

  store = new Vuex.Store({
    modules: {
      post
    }
  })

  wrapper = mount(Component, {
    localVue,
    store
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('getters', () => {
  it('post/spotFormData', () => {
    expect(wrapper.vm.spotFormData).toEqual(post.getters.spotFormData())
  })
})

describe('v-on', () => {
  it('postSpot', () => {
    const event = jest.fn()
    wrapper.setMethods({ postSpot: event })
    wrapper.find('.v-btn').trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })
})

describe('actions', () => {
  it('post/postSpot', () => {
    wrapper.vm.postSpot()
    expect(post.actions.postSpot).toHaveBeenCalled()
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
