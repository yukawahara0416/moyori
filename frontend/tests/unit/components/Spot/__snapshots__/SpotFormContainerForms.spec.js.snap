// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div
  class="container container--fluid"
>
  <div
    class="row align-center justify-center"
    style="flex-direction: column;"
  >
    <div
      class="col"
    >
      <div
        class="mx-5 v-card v-sheet theme--light"
      >
        <header
          class="white--text v-sheet v-sheet--tile theme--light v-toolbar v-toolbar--dense v-toolbar--flat primary"
          style="height: 48px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 48px;"
          >
            <div
              class="v-toolbar__title"
            >
              スポットを新規作成
            </div>
          </div>
        </header>
         
        <div
          class="v-card__text"
        >
          <form
            class="v-form"
            novalidate="novalidate"
          >
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-coffee theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="input-91"
                      style="right: 0px; position: absolute;"
                    >
                      スポット名
                    </label>
                    <input
                      id="input-91"
                      name="name"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--clear"
                    >
                      <button
                        aria-label="clear icon"
                        class="v-icon notranslate v-icon--link material-icons theme--light"
                        type="button"
                      >
                        $clear
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-home-circle-outline theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="input-96"
                      style="right: 0px; position: absolute;"
                    >
                      住所（クリックした座標から候補を表示しています）
                    </label>
                    <input
                      id="input-96"
                      name="address"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--clear"
                    >
                      <button
                        aria-label="clear icon"
                        class="v-icon notranslate v-icon--link material-icons theme--light"
                        type="button"
                      >
                        $clear
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-map-marker-outline theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="input-101"
                      style="right: 0px; position: absolute;"
                    >
                      イメージ
                    </label>
                    <input
                      id="input-101"
                      name="image"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--clear"
                    >
                      <button
                        aria-label="clear icon"
                        class="v-icon notranslate v-icon--link material-icons theme--light"
                        type="button"
                      >
                        $clear
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-web theme--light"
                  />
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="input-106"
                      style="right: 0px; position: absolute;"
                    >
                      URL
                    </label>
                    <input
                      id="input-106"
                      name="url"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--clear"
                    >
                      <button
                        aria-label="clear icon"
                        class="v-icon notranslate v-icon--link material-icons theme--light"
                        type="button"
                      >
                        $clear
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
         
        <div
          class="v-card__actions"
        >
          <div
            class="spacer"
          />
           
          <button
            class="mb-3 px-10 v-btn v-btn--contained theme--light v-size--large primary"
            type="submit"
          >
            <span
              class="v-btn__content"
            >
              
            スポットを登録する
          
            </span>
          </button>
           
          <div
            class="spacer"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
